File: C/2nd Sem/Module 2.1/7. factorial/factorial.c
#include<stdio.h>
void main()
{
    long int a=1, fact=1, n;
    printf("Enter a number\n");
    scanf("%ld",&n);
    while(a<=n)
    {
        fact=fact*a;
        a=a+1;
    }
    printf("factorial = %ld",fact);
}

File: Java/leetcode/recursion/Factorial.java
package recursion;

import java.util.Scanner;

public class Factorial
{
    public static void main(String[] args)
    {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter number: ");
        int n = in.nextInt();

        int ans = fact(n);
        System.out.println(ans);
    }

    private static int fact(int n)
    {
        if(n==1)
        {
            return 1;
        }
        return n*fact(n-1);
    }
}
